### PR TITLE
#652: feature: HRANDFIELD command

### LINKED ISSUE #652: Add support for command `HRANDFIELD`
Add support for the `HRANDFIELD` command in DiceDB similar to the [`HRANDFIELD` command in Redis](https://redis.io/docs/latest/commands/command-list/). Please refer to the following commit in Redis to understand the implementation specifics - [source](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa).

Write unit and integration tests for the command referring to the tests written in the [Redis codebase 7.2.5](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa). For integration tests, you can refer to the `tests` folder. Note: they have used `TCL` for the test suite, and we need to port that to our way of writing integration tests using the relevant helper methods. Please refer to our [tests directory](https://github.com/DiceDB/dice/tree/master/tests).

For the command, benchmark the code and measure the time taken and memory allocs using [`benchmem`](https://blog.logrocket.com/benchmarking-golang-improve-function-performance/) and try to keep them to the bare minimum.

### PR DESCRIPTION
**Pull request by Indranil0603**
Closes #652

Adds support for the **HRANDFIELD** command. This command returns one or more random fields from the hash stored at the specified key. If the "WITHVALUES" option is provided, both fields and their corresponding values are returned. It supports fetching a specific number of fields based on the count argument, with positive values returning unique fields and negative values allowing repeated selections.

Redis doc: **[HRANDFIELD](https://redis.io/docs/latest/commands/hrandfield/)**

Changes:

- Implementation - evalHRANDFIELD
- Unit tests - testEvalHRANDFIELD
- Integration tests - TestHRANDFIELD

### CODE REVIEW

**Comment by JyotinderSingh** [comment]
Could you please resolve the conflicts?

**Comment by Indranil0603** [comment]
Hi @JyotinderSingh  I have resolved the conflicts

File: internal/eval/tmp/main
Comment (lucifercr07) [inline internal/eval/tmp/main:1]: @Indranil0603 please remove this file

Comment (Indranil0603) [reply internal/eval/tmp/main:1]: Removed this file

**Review by JyotinderSingh — APPROVED** [review]
Thanks for adding support for `HRANDFIELD` command @Indranil0603! I've changed the structure of the tests and the eval implementation a little to make it more readable.

Please go through it for future reference.

**Pull request merged**